sdk/storage/src/table/cloud_table.rs

```diff
@@ -288,27 +322,86 @@ where
         let (headers, body) =
             check_status_extract_headers_and_body(future_response, StatusCode::OK).await?;
 
-        log::trace!("body == {:?}", std::str::from_utf8(&body));
-        let entities = serde_json::from_slice::<EntityCollection<T>>(&body)?;
-        *continuation = Continuation::try_from(&headers)?;
-        Ok(Some(entities.value))
+        Ok((continuation_token.query_path.as_str(), &headers, &body).try_into()?)
+    }
+
+    pub fn stream_get_all<'a, T>(
+        &'a self,
+    ) -> impl Stream<Item = Result<QueryResult<T>, AzureError>> + 'a
+    where
+        T: Serialize + DeserializeOwned + 'a,
+    {
+        #[derive(Debug, Clone)]
+        enum States {
+            Init,
+            Continuation(ContinuationToken),
+        };
+
+        futures::stream::unfold(
+            Some(States::Init),
+            move |state: Option<States>| async move {
+                log::debug!("state == {:?}", state);
+                let response = match state {
+                    Some(States::Init) => self.begin_get_all().await,
+                    Some(States::Continuation(continuation_token)) => {
+                        self.continue_execution(continuation_token).await
+                    }
+                    None => return None,
+                };
+
+                let response = match response {
+                    Ok(response) => response,
+                    Err(err) => return Some((Err(err), None)),
+                };
+
+                let continuation_token = match &response.continuation_token {
+                    Some(ct) => Some(States::Continuation(ct.to_owned())),
+                    None => None,
+                };
```

Comment:
```suggestion
                let continuation_token = response.continuation_token.map(|ct| States::Continuation(ct.to_owned()));
```

Reply:
I just had to slightly modify the code because `map` partially moves `response`:

```rust
let continuation_token = response
   .continuation_token
   .clone()
   .map(|ct| States::Continuation(ct));
```